### PR TITLE
adding a check to bootstrap script

### PR DESCRIPTION
**Pull request by ducks**
and a check to the rust config script

refs #36207 

first crack at making configure detect nodejs

### CODE REVIEW

**Comment by alexcrichton** [comment]
Thanks for the PR! The `nodejs` logic in `bootstrap.py` though seems to never be called? I think this'll want to perhaps live in `sanity.rs` or close to that where we do the probing. A manually configured `nodejs = "..."`, however, would override whatever we end up probing for.

**Comment by rjgoldsborough** [comment]
Yeah, I didn't make it as far as actually make it do something yet.

I was reading through the bootstrap readme and was probably ask for some guidance but thanks for the advance advice. :)

**Comment by brson** [comment]
I don't have a solid grasp on how all the build system pieces work these days, but I expected this detection to happin in `configure`, whereafter bootstrap would read it from `config.mk`, and in turn pass it to `compiletest`.

**Comment by rjgoldsborough** [comment]
Dang, ok. I was a bit off. I think I focused on /src/bootstrap too much and will look at configure more.

**Comment by rjgoldsborough** [comment]
So I think I made a *bit* of progress. I am now detecting nodejs back in `./configure`.
I then pass it through `config.rs` when it slurps up the `config.mk`. It seems to pass the sanity check that I added to `sanity.rs` because when running `make`, bootstrap compiles fine. I also did some println!ing and it was showing the correct node path. 

I get a bit lost after this. Where to pass it next or how to add a test to compiletest for it or if I need to? Any advice on where to look next?

Thanks in advance! :)

File: src/bootstrap/sanity.rs [hunk 1]
```diff
@@ -75,6 +75,12 @@ pub fn check(build: &mut Build) {
 
     need_cmd("python".as_ref());
 
+    // If a manual nodejs was added to the config,
+    // of if a nodejs install is detected through config, use it.
+    if build.config.nodejs.is_some() {
+        need_cmd("nodejs".as_ref())
```
Comment (alexcrichton) [inline src/bootstrap/sanity.rs]: Here I think you'll want to do:

```rust
if let Some(ref s) = build.config.nodejs {
    need_cmd(s.as_ref());
}
```

**Comment by alexcrichton** [comment]
@bors: r+ 89bc13c37d0fc58b583f07a5902f62e598c69b0e

Thanks!

**Pull request merged**